### ISSUE TITLE
NullReferenceException on QuickLogin

### ISSUE THREAD

**Issue by derwaldgeist** [lupidan/apple-signin-unity#132]
I'm not sure if this is related to the update to iOS 15. I just noticed that - without any changes to my app - `authManager.QuickLogin` fails, due to a crash in the `OnAfterSerialize` method in `AppleIdCredential.cs`:

```
NullReferenceException: Object reference not set to an instance of an object.
  at AppleAuth.Native.SerializationTools.FixSerializationForFullPersonName (AppleAuth.Native.FullPersonName& originalFullPersonName) [0x00000] in <00000000000000000000000000000000>:0 
  at AppleAuth.Native.AppleIDCredential.OnAfterDeserialize () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.JsonUtility.FromJson[T] (System.String json) [0x00000] in <00000000000000000000000000000000>:0 
  at AppleAuth.AppleAuthManager+<>c__DisplayClass7_0.<QuickLogin>b__0 (System.String payload) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Action`1[T].Invoke (T obj) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Action.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at AppleAuth.AppleAuthManager+CallbackHandler.ExecutePendingCallbacks () [0x00000] in <00000000000000000000000000000000>:0 
  at Marble.Authentication.AppleAuthenticationManager+<CheckAppleAuthStatus>d__10.MoveNext () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00000] in <00000000000000000000000000000000>:0 
```

which then subsequently causes my own code to crash with another NRE. Because IL2CPP won't show the line numbers, I can only guess this happens when I access  a property of the appleIdCredential (most likely the FullName, which I then compare to null, as I am aware it won't be set on QuickLogin).

**Comment by lupidan** [comment]
Hi. It´s some strange behaviour that Apple decided to start doing. When not requesting the name, Apple started sending an instance of NSPersonNameComponents with no strings in it 🤷 

This was reported some time ago:
* https://github.com/lupidan/apple-signin-unity/issues/110

And a fix for it was published soon after that:
* https://github.com/lupidan/apple-signin-unity/releases/tag/v1.4.2

Updating to v1.4.2 should work. 👍

**Comment by derwaldgeist** [comment]
Hey, thanks for the super-fast response. Yes, I just realized that when I looked at the change log. I tried it in the meantime, and it fixed the NRE in the serializer. However, my own code still crashes when trying to access one of the properties after login. Trying to find out which one it is (just added some debug logs before each line; IL2CPP really sucks).

**Comment by derwaldgeist** [comment]
I've now checked my code step by step. The NRE is thrown because `appleIdCredential.AuthorizationCode` is `null`. However, I need these credentials to login to my app. What might be wrong here?

I also noticed that - if that happens - the same problem occurs if I then try to login with the regular login, i.e. without using QuickLogin. *Only* if I restart the app completely, skip the QuickLogin and then use the regular Apple login, the `AuthorizationCode` returns a value.
